packages/deployment/lib/kes.js

```diff
@@ -151,14 +152,16 @@ class UpdatedKes extends Kes {
     const src = path.join(process.cwd(), kesBuildFolder, filename);
     const dest = path.join(process.cwd(), kesBuildFolder, 'adapter', unzipFolderName);
 
-    // return Promise.resolve();
+    if (!filename) return super.compileCF();
     return fetchMessageAdapter(
       this.config.message_adapter_version,
       this.messageAdapterGitPath,
       filename,
       src,
       dest
-    ).then(() => super.compileCF());
+    ).then(() => {
+      this.Lambda.messageAdapterZipFileHash = new this.Lambda(this.config).getHash(src);
```

Comment:
@jennyhliu the problem with this approach is that the hash only changes if the message adapter is changed. this means that if you make a change to your lambda function, it never gets updated.

Reply:
The hash of the message adapter is prepend to the lambda zip file name, so one of the hash values is changed, the lambda zip file name is changed, the zipLambda will create a new zip file. e.g.
Zipping app/build/cloudformation/da2198a240ed8948234f5e462236cbc57a6fed49-2354f9a4959ba95c1e7c8d12a09f5c09e573749a-ParsePdr.zip and injecting message adapter for ParsePdr
When I updated the existing zip file name.

Reply:
oh, right. so the lambda name already has the hash in it. didn't realize that. should be worried about the filename length?

Reply:
S3 Max Key Length is 1024, and mac file name limit is 255.  We should be fine.  We can change to use cksum if it becomes an issue in the future.